infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutGeometry.kt

```diff
@@ -100,11 +100,16 @@ interface IAlignment : Loggable {
         else if (m >= length) end
         else getSegmentAtM(m)?.seekPointAtM(m, snapDistance)?.point
 
-    fun getSegmentIndexAtM(m: Double) = segments.binarySearch { s ->
-        if (m < s.startM) 1
-        else if (m > s.endM) -1
-        else 0
-    }
+    fun getSegmentIndexAtM(m: Double) =
+        if (m < 0 || m > length + LAYOUT_M_DELTA) throw IllegalArgumentException("m of $m out of range 0..${length + LAYOUT_M_DELTA}") else m
```

Comment:
Pitäisiköhän tuon deltan verran heittoa sallia myös 0:sta alaspäin, ihan yhtenäisyyden vuoksi. Normaalistihan 0.0 säilyy toki sellaisenaan, joten lähinnä tulisi kyseeseen jos m-arvo olisi saatu jotenkin laskemalla nollaksi.